exercises/hamming/test/Tests.hs

```diff
@@ -102,3 +110,65 @@ cases = [ Case { description = "empty strands"
                , expected    = Nothing
                }
         ]
+
+hammingGen :: Int -> Gen Case
+hammingGen 0 = return
+  Case { description = "Empty strands have no distance"
+       , strand1 = ""
+       , strand2 = ""
+       , expected = Just 0
+       }
+
+hammingGen n = do
+  (a, b, count) <- nucleotidePairGen
+  case' <- hammingGen (n-1)
+  return . hammingDesc $
+    case' { description = ""
+          , strand1 = a ++ strand1 case'
+          , strand2 = b ++ strand2 case'
+          , expected = (+) <$> count <*> expected case'
+          }
+
+hammingDesc :: Case -> Case
+hammingDesc case'@Case{..} = case' { description = desc' }
+  where
+    desc' = "Strands of " ++ length' ++ " with " ++ dist' ++ " distance"
+    dist' = maybe "invalid" show expected
+    length' = case comparing length strand1 strand2 of
+      EQ -> "lengths " ++ show (length strand1)
+      _  -> "different lengths " ++ show (length strand1) ++ " and " ++
+            show (length strand2)
+
+nucleotidePairGen :: Gen (String, String, Maybe Integer)
+nucleotidePairGen = frequency
+  [ (20, zeroDistance)
```

Comment:
any particular reason these particular relative frequencies were chosen? Arbitrary, or was there some driving principle?

In either case, I'd suggest a code comment saying it, so that a next reader of the code will better understand if it is okay to change it to other values

Reply:
They were chosen "experimentally" such that the probability of producing a negative input is somewhat low. But at this point I've decided that I should rather decide from the top whether it is a positive or a negative case; the current approach felt easy and interesting, but the probabilities are not clear or ideal. (For example, in increasing the size parameter, negative cases will dominate.)

Reply:
I will look into this in some weeks.